cmd/krew/cmd/update.go

```diff
@@ -17,15 +17,31 @@ package cmd
 import (
 	"fmt"
 	"os"
+	"path/filepath"
+	"sort"
 
 	"github.com/pkg/errors"
 	"github.com/spf13/cobra"
 	"k8s.io/klog"
 
 	"sigs.k8s.io/krew/internal/gitutil"
+	"sigs.k8s.io/krew/internal/index/indexscanner"
+	"sigs.k8s.io/krew/internal/installation"
 	"sigs.k8s.io/krew/pkg/constants"
 )
 
+type newPlugin struct {
```

Comment:
I feel like you can simply have 1 type to carry out everything

```
type pluginInfo struct {
    name, version string
}
```

If you simply return `[]pluginInfo` from everywhere, when the time comes to printing, just diff two `[]pluginInfo` slices to find the "new" and "updated" plugins.

Correct me if I'm wrong but this can even help reduce some of the code.